bundles/org.openhab.binding.knx/src/main/java/org/openhab/binding/knx/internal/client/DeviceInspector.java

```diff
@@ -102,15 +104,128 @@ private Map<String, String> readDeviceProperties(IndividualAddress address) {
                 Thread.sleep(OPERATION_INTERVAL);
                 String manufacturerID = Manufacturer.getName(toUnsigned(getClient().readDeviceProperties(address,
                         DEVICE_OBJECT, PID.MANUFACTURER_ID, 1, 1, false, OPERATION_TIMEOUT)));
+
                 Thread.sleep(OPERATION_INTERVAL);
                 String serialNo = toHex(getClient().readDeviceProperties(address, DEVICE_OBJECT, PID.SERIAL_NUMBER, 1,
                         1, false, OPERATION_TIMEOUT), "");
+
                 Thread.sleep(OPERATION_INTERVAL);
                 String hardwareType = toHex(getClient().readDeviceProperties(address, DEVICE_OBJECT, HARDWARE_TYPE, 1,
                         1, false, OPERATION_TIMEOUT), " ");
+
+                // PID_FIRMWARE_REVISION, optional, fallback PID_VERSION according to spec
+                Thread.sleep(OPERATION_INTERVAL);
+                String firmwareRevision = null;
+                try {
+                    byte[] result = getClient().readDeviceProperties(address, DEVICE_OBJECT, PID.FIRMWARE_REVISION, 1,
+                            1, false, OPERATION_TIMEOUT);
+                    if (result != null) {
+                        firmwareRevision = Integer.toString(toUnsigned(result));
+                    } else {
+                        // try fallback to PID_VERSION
+                        result = getClient().readDeviceProperties(address, DEVICE_OBJECT, PID.VERSION, 1, 1, false,
+                                OPERATION_TIMEOUT);
+                        if (result != null) {
+                            // data format is DPT217.001
+                            int i = toUnsigned(result);
+                            firmwareRevision = Integer.toString((i & 0xF800) >> 11) + "."
+                                    + Integer.toString((i & 0x07C0) >> 6) + "." + Integer.toString((i & 0x003F));
+                        }
+                    }
+                } catch (Exception e) {
```

Comment:
No chance to catch more specialized exceptions ?
You could also name the variable `ignore` instead of `e`
Same for others

Reply:
I don't mind about what exception is thrown. These are optional properties, any failure should result in skipping it and continue reading the next one.
But thanks, I added catching and re-throwing ```InterruptedException``` where applicable.

Reply:
I do not understand all your logic with `InterruptedException`, can you please explain it ?
Normally, this kind of exception should not be caught, or if caught you should at least "retag" the thread as interrupted.
Is it a documented exception for `getClient().readDeviceProperties` ?
And I see that sometimes you consider it (by catching and rethrowing it), sometimes you ignore it (by catching any `Exception`). What is your general logic ?

Reply:
@lolodomo 
The concept should be as follows:
Thing creation (and status change of bridge to online) triggers several scheduled tasks, including periodic datapoint reads, reading the device info, etc. This scheduled tasks are cancelled once the thing is disconnected, thus ```InterruptedException``` is thrown. All tasks should end immediately (of course this will only happen at certain points of the code, e.g. sleep).

Reading the device properties is some kind of optional task, and singe properties might not be available. This is why exceptions are ignored and the binding tries to continue with the next property. Anyhow, interruption should stop the whole task.

Catching ```InterruptedException``` was actually targeting ```Thread.sleep(OPERATION_INTERVAL);```. ```readDeviceProperties``` does not throw ```InterruptedException```, it silently fails by catching it and returning null.
See DeviceInfoClientImpl.java:70.

Actually this should be corrected. Give me some time.

Reply:
@lolodomo does it look more reasonable now? I did the change described above and added javadoc.